### PR TITLE
Translations: Add a checkbox for case-sensitive search

### LINKED ISSUE #312: Translations: Include check button for case-sensitive search
From https://glotpress.trac.wordpress.org/ticket/151 by @zodiac1978:

> If you search for something like this: "Wordpress" And you want to replace it with "WordPress" (p -> P) you found a lot of crap you didn't looking for. A check button for switching on case sensitive search would be helpful.

### PR DESCRIPTION
**Pull request by ocean90**
Fixes #312.

Background: http://dev.mysql.com/doc/refman/5.7/en/charset-binary-op.html

before

after

### CODE REVIEW

File: gp-templates/translations.php [hunk 1]
```diff
@@ -116,7 +116,8 @@
 		</dd>
 		<dd>
 			<input type="checkbox" name="filters[with_comment]" value="yes" id="filters[with_comment][yes]" <?php gp_checked( 'yes' == gp_array_get( $filters, 'with_comment' ) ); ?>><label for='filters[with_comment][yes]'><?php _e( 'With comment' ); ?></label><br />
-			<input type="checkbox" name="filters[with_context]" value="yes" id="filters[with_context][yes]" <?php gp_checked( 'yes' == gp_array_get( $filters, 'with_context' ) ); ?>><label for='filters[with_context][yes]'><?php _e( 'With context' ); ?></label>
+			<input type="checkbox" name="filters[with_context]" value="yes" id="filters[with_context][yes]" <?php gp_checked( 'yes' == gp_array_get( $filters, 'with_context' ) ); ?>><label for='filters[with_context][yes]'><?php _e( 'With context' ); ?></label><br />
+			<input type="checkbox" name="filters[case_sensitive]" value="yes" id="filters[case_sensitive][yes]" <?php gp_checked( 'yes' == gp_array_get( $filters, 'case_sensitive' ) ); ?>><label for='filters[case_sensitive][yes]'><?php _e( 'Case sensitive' ); ?></label>
```
Comment (toolstack) [inline gp-templates/translations.php]: Shouldn't the `_e()` include the text domain?

File: gp-includes/things/translation.php [hunk 2]
```diff
@@ -120,10 +120,11 @@ public function for_translation( $project, $translation_set, $page, $filters = a
 		$sort_by = gp_array_get( $sort_bys, gp_array_get( $sort, 'by' ),  gp_array_get( $sort_bys, $default_sort['by'] ) );
 		$sort_hows = array('asc' => 'ASC', 'desc' => 'DESC', );
 		$sort_how = gp_array_get( $sort_hows, gp_array_get( $sort, 'how' ), gp_array_get( $sort_hows, $default_sort['how'] ) );
+		$collation = 'yes' === gp_array_get( $filters, 'case_sensitive' ) ? 'BINARY' : '';
 
 		$where = array();
 		if ( gp_array_get( $filters, 'term' ) ) {
-			$like = "LIKE '%" . ( esc_sql( $wpdb->esc_like( gp_array_get( $filters, 'term' ) ) ) ) . "%'";
+			$like = "LIKE $collation '%" . ( esc_sql( $wpdb->esc_like( gp_array_get( $filters, 'term' ) ) ) ) . "%'";
```
Comment (toolstack) [inline gp-includes/things/translation.php:127]: Should the extra space be in the $collation value so we don't get a double space after the LIKE?

Comment (ocean90) [inline gp-includes/things/translation.php:127]: We could, but there is no harm in having the extra space there. For readability I'm going to leave it as is.

File: gp-templates/translations.php (hunk 1 above)
Comment (ocean90) [inline gp-templates/translations.php]: Yes, fixed in 22d0ad0.

**Comment by toolstack** [comment]
:+1:

**Comment by Zodiac1978** [comment]
:+1:

**Pull request merged**